processing/src/main/java/io/druid/query/aggregation/post/ExpressionPostAggregator.java

```diff
@@ -0,0 +1,195 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.post;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+import io.druid.math.expr.Expr;
+import io.druid.math.expr.Parser;
+import io.druid.query.aggregation.PostAggregator;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ */
+public class ExpressionPostAggregator implements PostAggregator
+{
+  private static final Comparator<Number> DEFAULT_COMPARATOR = new Comparator<Number>()
+  {
+    @Override
+    public int compare(Number o1, Number o2)
+    {
+      if (o1 instanceof Long && o2 instanceof Long) {
+        return Long.compare(o1.longValue(), o2.longValue());
+      }
+      return Double.compare(o1.doubleValue(), o2.doubleValue());
+    }
+  };
+
+  private final String name;
+  private final String expression;
+  private final Comparator comparator;
+  private final String ordering;
+
+  private final Expr parsed;
+  private final List<String> dependentFields;
+
+  @JsonCreator
+  public ExpressionPostAggregator(
+      @JsonProperty("name") String name,
+      @JsonProperty("expression") String expression,
+      @JsonProperty("ordering") String ordering
+  )
+  {
+    Preconditions.checkArgument(expression != null, "expression cannot be null");
+
+    this.name = name;
+    this.expression = expression;
+    this.ordering = ordering;
+    this.comparator = ordering == null ? DEFAULT_COMPARATOR : Ordering.valueOf(ordering);
+
+    this.parsed = Parser.parse(expression);
+    this.dependentFields = Parser.findRequiredBindings(parsed);
+  }
+
+  public ExpressionPostAggregator(String name, String fnName)
+  {
+    this(name, fnName, null);
+  }
+
+  @Override
+  public Set<String> getDependentFields()
+  {
+    return Sets.newHashSet(dependentFields);
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return comparator;
+  }
+
+  @Override
+  public Object compute(Map<String, Object> values)
+  {
+    return parsed.eval(Parser.withMap(values));
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty("expression")
+  public String getExpression()
+  {
+    return expression;
+  }
+
+  @JsonProperty("ordering")
+  public String getOrdering()
+  {
+    return ordering;
+  }
+
+  @Override
+  public String toString()
+  {
+    return "ExpressionPostAggregator{" +
+           "name='" + name + '\'' +
+           ", expression='" + expression + '\'' +
+           ", ordering=" + ordering +
+           '}';
+  }
+
+  public static enum Ordering implements Comparator<Number>
```

Comment:
Ah ha, I knew I saw this somewhere before :)

Yeah, I think we could have a set of `NumberComparators` utilities sort of like we have `StringComparators`. Separate PR would be fine by me though.